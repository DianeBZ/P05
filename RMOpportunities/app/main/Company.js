import React, { Component } from 'react';
import { Text, Image, StyleSheet, View,ScrollView, Dimensions, TouchableHighlight, BackAndroid} from 'react-native';

import Translation from './Translation';

// Variable used to decide which information is rendered
var width = Dimensions.get('window').width;
var height = Dimensions.get('window').height;

// Variable used to decide which information is rendered
// if about is true, we display company information
// if about is false, we display team information
var about = 'false';

class BlockFunction extends Component{
 render(){
     return(
       <View>
       {(() => {
           if (about==='true'){
             return(
			 <View>
				<Text style={styles.titleCompany2}>{Translation[lang].lentreprise}</Text>
				<TouchableHighlight onPress={this._onPressAbout}>
					<View style={styles.titleStyle}>
						<Text style={styles.title}>{Translation[lang].a_propos}</Text>
					</View>
				</TouchableHighlight>
				<View style={styles.separation}></View>
				<View style={styles.about}>
					<Text style={styles.aboutText}>{Translation[lang].a_propos1}</Text>

					<Text style={styles.aboutText}>{Translation[lang].a_propos2}</Text>

					<Text style={styles.aboutText}>{Translation[lang].a_propos3}</Text>

					<Text style={styles.aboutText}>{Translation[lang].a_propos4}</Text>
				</View>
				<TouchableHighlight onPress={this._onPressAbout}>
					<View style={styles.titleStyle}>
						<Text style={styles.title}>{Translation[lang].equipe}</Text>
					</View>
				</TouchableHighlight>
				<View style={styles.separation}></View>
			</View>
			 );
			}else{
             return(
			 <View>
				<Text style={styles.titleCompany2}>{Translation[lang].lentreprise}</Text>
				<TouchableHighlight onPress={this._onPressAbout}>
					<View style={styles.titleStyle}>
						<Text style={styles.title}>{Translation[lang].a_propos}</Text>
					</View>
				</TouchableHighlight>
				<View style={styles.separation}></View>
				<TouchableHighlight onPress={this._onPressAbout}>
					<View style={styles.titleStyle}>
						<Text style={styles.title}>{Translation[lang].equipe}</Text>
					</View>
				</TouchableHighlight>
				<View style={styles.separation}></View>
				<View style={styles.containerV}>
					<View style={styles.element1}>
						<Image source={require('../img/Gregoire.jpg')} style = {styles.picture} />
					</View>
					<View style={styles.element2}>
						<Text style={styles.name}>Grégoire Willmann</Text>
						<Text style={styles.teamText}> {Translation[lang].statutG}</Text>
						<Text style={styles.teamText}>{Translation[lang].diplome}</Text>
					</View>
				</View>
				<View style={styles.containerV}>
					
					<View style={styles.element2}>
						<Text style={styles.name}>Gabriel Calmels</Text>
						<Text style={styles.teamText}> {Translation[lang].statutGa}</Text>
						<Text style={styles.teamText}>{Translation[lang].diplome}</Text>
					</View>
					<View style={styles.element1}>
						<Image source={require('../img/Gabriel.jpg')} style = {styles.picture} />
					</View>
				</View>
				<View style={styles.containerV}>
					<View style={styles.element1}>
						<Image source={require('../img/Christian.png')} style = {styles.picture} />
					</View>
					<View style={styles.element2}>
						<Text style={styles.name}>Christian Rodié</Text>
						<Text style={styles.teamText}>{Translation[lang].statutC}</Text>
						<Text style={styles.teamText}>{Translation[lang].exPoste}</Text>
					</View>
				</View>
				</View>
			 );
			}
		})()}
   </View>
 );
 }
 // function used to change value of variables (touching title)
_onPressAbout = () =>{
 if(about === 'true'){
   about = 'false';
 }else{
   about ='true';
 }
 this.forceUpdate();
};

}
export default class Company extends Component {
  render() {
    BackAndroid.addEventListener('Back',this.onBackAndroid);
    return (
      <ScrollView>
		
		<BlockFunction/>
     </ScrollView>
    );
  }
  onBackAndroid = () => {
		const { navigator } = this.props;
		if (navigator && navigator.getCurrentRoutes().length > 1) {
			BackAndroid.removeEventListener('Back', this.onBackAndroid);
			navigator.pop();
			return true;
		} else {
			return false;
		}
  };
}

const styles = StyleSheet.create({
	titleCompany:{
		fontSize:30,
		margin:width*0.2,
		textAlign:'center',
	},
	titleCompany2:{
		fontSize:30,
		marginTop:width*0.2,
		marginBottom:width*0.05,
		textAlign:'center',
	},
	titleStyle:{
	},
	title:{
		color:'#A4D04A',
		marginLeft:width*0.05,
		fontSize:22,
		fontStyle:'italic',
		fontWeight: 'bold',
	},
	about:{
		margin:width*0.03,
	},
	aboutText:{
		fontSize:17,
		marginBottom:width*0.03,
	},
	
	separation:{
      borderTopWidth: 2,
      marginTop:10,
      marginBottom:30,
      marginRight:width*0.05,
      marginLeft:width*0.05,
      borderTopColor:"#A4D04A",
   },
   containerV:{
		flexDirection: 'row',
		marginLeft:width*0.03,
		marginRight:width*0.03,
   },
   element1:{
	   width: width*0.3,
		height:width*0.3,
   },
   element2:{
		width: width*0.6,
		height:width*0.3,
		margin:width*0.01,
		alignItems:'center',
		justifyContent: 'center',
   },
  picture:{
    width: width*0.3,
    height: width*0.3,
  },
  name:{
	  textAlign:'center',
		fontWeight: "bold",	  
  },
  teamText:{
	textAlign:'center',  
  },
});