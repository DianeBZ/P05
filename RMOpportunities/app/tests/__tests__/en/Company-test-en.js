// __tests__/Company-test-en.js
import 'react-native';
import React from 'react';

global.lang = 'en';
import Company from '../../../main/Company';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <Company />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
