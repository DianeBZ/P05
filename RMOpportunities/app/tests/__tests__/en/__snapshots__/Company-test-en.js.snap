// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "height": 106.72,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#F2F2F2",
        }
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000000",
              "fontSize": 32,
              "marginBottom": 10,
              "marginTop": 15,
              "textAlign": "center",
            }
          }
        >
          The company
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#A4D04A",
                  "fontSize": 22,
                  "fontStyle": "italic",
                  "fontWeight": "bold",
                  "marginLeft": 37.5,
                }
              }
            >
              About
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "borderTopColor": "#A4D04A",
              "borderTopWidth": 2,
              "marginBottom": 30,
              "marginLeft": 37.5,
              "marginRight": 37.5,
              "marginTop": 10,
            }
          }
        />
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#A4D04A",
                  "fontSize": 22,
                  "fontStyle": "italic",
                  "fontWeight": "bold",
                  "marginLeft": 37.5,
                }
              }
            >
              The team
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "borderTopColor": "#A4D04A",
              "borderTopWidth": 2,
              "marginBottom": 30,
              "marginLeft": 37.5,
              "marginRight": 37.5,
              "marginTop": 10,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginLeft": 22.5,
              "marginRight": 22.5,
            }
          }
        >
          <View
            style={
              Object {
                "height": 225,
                "width": 225,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 225,
                  "width": 225,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 225,
                "justifyContent": "center",
                "margin": 7.5,
                "width": 450,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Grégoire Willmann
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
               
              Co-founder & President
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Engineers who graduated from Telecom Saint-Etienne
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginLeft": 22.5,
              "marginRight": 22.5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 225,
                "justifyContent": "center",
                "margin": 7.5,
                "width": 450,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Gabriel Calmels
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
               
              Co-founder & CEO
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Engineers who graduated from Telecom Saint-Etienne
            </Text>
          </View>
          <View
            style={
              Object {
                "height": 225,
                "width": 225,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 225,
                  "width": 225,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginLeft": 22.5,
              "marginRight": 22.5,
            }
          }
        >
          <View
            style={
              Object {
                "height": 225,
                "width": 225,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 225,
                  "width": 225,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 225,
                "justifyContent": "center",
                "margin": 7.5,
                "width": 450,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Christian Rodié
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Administrator
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Ex-CEO of ISOCHEM
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
